package/MDAnalysis/analysis/dielectric.py

```diff
@@ -65,6 +65,10 @@ class DielectricConstant(AnalysisBase):
     the usual case if electrostatics are handled with a Ewald summation
     technique. See [Neumann1983]_ for details on the derivation.
 
+    .. warning::
+      Applying this class requires that no free charges, such as ions or
```

Comment:
Maybe a stupid question, if this is relying on an Ewald style summation, how well does it handle a system with a net charge?

Reply:
No, totally valid question. The answer will be a bit longer. sorry.

Ewald can handle systems with a net charge. You introduce a homogenous background charge that neutralizes the system, which will lead to an additional force an all atoms. This is unproblematic for homogenous systems but [lead to severe artifacts in inhomogenous systems like membranes etc](http://dx.doi.org/10.1021/ct400626b).

The equation we use here is derived for an Ewald boundary condition, meaning that the dielectric constant of sorrounding medium is a metal ($\varepsilon=\infty$).  If you have other boundary conditions like when using reaction field or cutoff electrostatics the equation will change as described in the paper we cite. We use eq 26.4.3D because the derivation for an Ewald electrostatic leads to the same results as a reaction field with an dielectric constant of $\infty$.

<img width="845" alt="Screenshot 2023-08-28 at 21 39 54" src="https://github.com/MDAnalysis/mdanalysis/assets/15192133/66b30258-4e1e-4a08-97b3-c45966f817c9">

Now, why are charges a problem? The issue is that in the derivation you demand that Maxwell's equation for electric field

$$
\nabla \vec E = \frac{\rho}{\varepsilon_0}
$$

where $\rho$ is the charge density is equal to zero: $\nabla \vec E = 0$. This means that no free charges are present. If you do not do this you will not end up with the equation we use.

Does this answer your question?

Reply:
Do you think you could transfer this very insightful piece of writing to an issue for future reference. :)

Reply:
Yes we sure. If people find it there.